File: football-backend/src/main/java/com/testowanie/football/dto/resource/ArticleResource.java
package com.testowanie.football.dto.resource;

import lombok.Builder;
import lombok.extern.jackson.Jacksonized;

import java.time.Instant;
import java.util.List;

@Builder
@Jacksonized
public record ArticleResource(
        Long id,
        String title,
        EditorResource editor,
        Instant createdDate,
        Instant lastModifiedDate,
        String content,
        String image,
        CategoryResource category,
        List<CommentResource> comments
) {
}

File: football-frontend/src/app/models/article-resource.ts
import { Comment } from './comment';
import { EditorResource } from './editor-resource';
import {CategoryResource} from "./category-resource";

export class ArticleResource {
    constructor(
        public id: number,
        public title: string,
        public editor: EditorResource,
        public createdDate: Date,
        public lastModifiedDate: Date,
        public content: string,
        public image: string,
        public category: CategoryResource,
        public comments: Comment[],
    ) {}
}
